libutils/json-utils.c

```diff
@@ -286,17 +326,29 @@ bool JsonParseCsvFile(const char *input_path, size_t size_max, JsonElement **jso
     }
 
     bool reached_eof = feof(fin);
-    fclose(fin);
 
     if (!reached_eof && byte_count <= size_max)
     {
         Log(LOG_LEVEL_ERR,
             "%s: unable to read line from CSV file '%s'. (fread: %s)",
             myname, input_path, GetErrorStr());
         JsonDestroy(json);
+        fclose(fin);
         return false;
     }
+    else if (JsonLength(json) == 0)
+    {
+        Log(LOG_LEVEL_WARNING,
+            "%s: CSV file '%s' is not empty, but nothing was parsed",
+            myname, input_path);
+        if (contains_newline(fin))
+        {
+            Log(LOG_LEVEL_WARNING,
+                "Make sure the file contains DOS (CRLF) line endings");
```

Comment:
*Make sure the file uses CRLF sequence as record separators (line endings)*

Reply:
I think karl's wording is easier to understand.